packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx

```diff
@@ -123,9 +106,156 @@ export class Tooltip {
+      case "bottom-start":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.left - dialogEl.left}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.bottom - dialogEl.top}px`
+        );
+        break;
+      case "bottom-end":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.right - dialogEl.right}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.bottom - dialogEl.top}px`
+        );
+        break;
+      case "top":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.left - dialogEl.left - 0.5 * child.width}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.top - dialogEl.bottom}px`
+        );
+        break;
+      case "top-start":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.left - dialogEl.left}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.top - dialogEl.bottom}px`
+        );
+        break;
+      case "top-end":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.right - dialogEl.right}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.top - dialogEl.bottom}px`
+        );
+        break;
+      case "left":
+      case "left-start":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.right - dialogEl.right - child.width}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.bottom - dialogEl.top - child.height}px`
+        );
+        break;
+      case "left-end":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.right - dialogEl.right - child.width}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.top - dialogEl.bottom + child.height}px`
+        );
+        break;
+      case "right":
+      case "right-start":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.left - dialogEl.left + child.width}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.bottom - dialogEl.top - child.height}px`
+        );
+        break;
+      case "right-end":
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_X_VARIABLE,
+          `${child.left - dialogEl.left + child.width}px`
+        );
+        this.toolTip.style.setProperty(
+          this.TOOLTIP_Y_VARIABLE,
+          `${child.top - dialogEl.bottom + child.height}px`
+        );
+        break;
+    }
+  };
+
```

Comment:
rather than lots of calls to this.toolTip.style.setProperty(....), could we just create local variables that get set inside the case: statements, then just have 2 calls to setProperty to set the x & y